packages/SwingSet/src/controller/initializeKernel.js

```diff
@@ -25,6 +43,9 @@ export async function initializeKernel(config, kernelStorage, options = {}) {
   const logStartup = verbose ? console.debug : () => 0;
   insistStorageAPI(kernelStorage.kvStore);
 
+  const CURRENT_VERSION = 1;
+  kernelStorage.kvStore.set('version', `${CURRENT_VERSION}`);
```

Comment:
Yeah, that sounds like the best plan. I'm not jazzed about the idea of `makeKernelKeeper()` returning an API surface for the wrong data (e.g. if `kernel.js` forgot to make the `getInitialized`/`assertInitialized`/`assertVersion` check, and started using a non-upgraded DB), but the real problem is that that we need a kernelKeeper to populate the initial state, which of course happens before the DB initialized.

Hm, what about `makeKernelKeeper(expectedVersion)` ? We'd use `undefined` for the initial construction, then the normal `kernel.js` usage would pass in `CURRENT_VERSION` or whatever. Oh, and what if we used `'uninitialized'` instead of `undefined`? That would make it impossible to call `makeKernelKeeper()`, and you'd have to be explicit about whether you're making one for initialization purposes, or for actual use?

Hm, now *that* makes me wonder if `upgradeSwingset` could go back to being a method of `kernelKeeper`, as I first tried. If we went that way, `upgradeSwingset()` would be like `makeKernelKeeper(..).upgrade()` .. hm, no, we wouldn't know what version to expect (it could be either 0 or 1, but shouldn't be missing entirely / uninitialized).

Reply:
Ok I went with `makeKernelKeeper(expectedVersion)`, and I like how it flows. `kernelKeeper.js` exports the current version, so the kernel demands it, and won't get back an API that's using wrong-version data. And there's a way for `initializeKernel` to demand an uninitialized one. The host app doesn't create the kernelKeeper, so it doesn't need to know anything about versions, it just calls `upgradeSwingset` (either every boot, or every boot after the operator/host-app-author knows they've changed the kernel code).